library/src/main/java/io/ona/kujaku/wmts/model/WmtsLayer.java

```diff
@@ -0,0 +1,163 @@
+
+import javax.annotation.Nullable;
+
+import io.ona.kujaku.wmts.model.common.ows.LanguageStringType;
+
+/**
+ * Describes a Wmts Layer object from the WMTS Capabilities object
+ *
+ * Created by Emmanuel Otin - eo@novel-t.ch 11/28/18.
+ */
+public class WmtsLayer {
+
+    @ElementList(inline=true, entry="Title")
+    private List<LanguageStringType> titles;
+
+    @Element(name="Identifier")
+    private String identifier;
+
+    @ElementList(inline=true, entry="Style")
+    private List<WmtsStyle> styles;
+
+    @ElementList(inline=true, entry="TileMatrixSetLink")
+    private List<WmtsTileMatrixSetLink> tileMatrixSetLinks;
+
+    @ElementList(inline=true, entry="ResourceURL")
+    private List<WmtsResourceUrl> resourceURLs;
+
+    private String selectedStyleIdentifier;
+
+    private String selectedTileMatrixLinkIdentifier;
+
+    private int maximumZoom;
+
+    private int minimumZoom;
+  
+    private int tilesSize;
+
+    @NonNull
+    public List<LanguageStringType> getTitles() {
+        return this.titles;
+    }
+
+    @NonNull
+    public String getIdentifier() {
+        return this.identifier;
+    }
+
+    public void setSelectedStyleIdentifier(@NonNull String selectedStyleIdentifier) {
+        this.selectedStyleIdentifier = selectedStyleIdentifier;
+    }
+
+    public void setSelectedTileMatrixLinkIdentifier(@NonNull String selectedTileMatrixLinkIdentifier) {
+        this.selectedTileMatrixLinkIdentifier = selectedTileMatrixLinkIdentifier;
+    }
+
+    @Nullable
+    public WmtsStyle getStyle(@NonNull String styleIdentifier) {
+        for (WmtsStyle style : this.styles) {
+            if (style.getIdentifier().equals(styleIdentifier)) {
+                return style;
+            }
+        }
+
+        return null ;
+    }
+
+    @Nullable
+    public WmtsTileMatrixSetLink getTileMatrixSetLink(@NonNull String tileMatrixLinkIdentifier) {
+        for (WmtsTileMatrixSetLink tileMatrixSetLink : this.tileMatrixSetLinks) {
+            if (tileMatrixSetLink.getTileMatrixSet().equals(tileMatrixLinkIdentifier)) {
+                return tileMatrixSetLink;
+            }
+        }
+
+        return null ;
+    }
+
+    @Nullable
+    public String getTemplateUrl(@NonNull String resourceType) {
+        WmtsResourceUrl url = this.getResourceUrl(resourceType);
+
+        if (url == null) {
+            return null;
+        }
+
+        if (this.selectedStyleIdentifier == null || this.selectedStyleIdentifier.isEmpty()) {
+            // Search for default style
+            WmtsStyle style = this.getDefaultStyle();
+            if (style == null) {
+               // search for the first available style
+                this.selectedStyleIdentifier = this.styles.get(0).getIdentifier();
+            } else {
+                this.selectedStyleIdentifier = style.getIdentifier();
+            }
+        }
+
+        return url.getTemplate(this.selectedStyleIdentifier, this.getSelectedTileMatrixLinkIdentifier());
+    }
+
+    @Nullable
```

Comment:
Sorry, seems that I made a mistake. This seems that it would work with `@NonNull`